pkg/apis/v1beta1/ec2nodeclass.go

```diff
@@ -62,8 +62,12 @@ type EC2NodeClassSpec struct {
 	// for the old instance profiles on an update.
 	// +kubebuilder:validation:XValidation:rule="self != ''",message="role cannot be empty"
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="immutable field changed"
-	// +required
-	Role string `json:"role"`
+	// +optional
+	Role string `json:"role,omitempty"`
```

Comment:
A string being optional doesn't mean that it needs to be a pointer. This basically just means that there's an expectation that empty string is invalid. Realistically, I think we should move to have more of our fields look like this in cases where an empty string is not a valid value.

Reply:
Can we make the instance profile name a non-pointer then too?

Reply:
I'm fine changing it if this change was only applying to version after v0.32.x but since this is going to get back-ported onto a patch version that is doing conversion from the v1alpha1 field which is a pointer, I'd rather not mess with it at this point since you lose information converting back between a pointer and a raw value.

Reply:
Good call.